packages/@stimulus/core/src/target_observer.ts

```diff
@@ -0,0 +1,50 @@
+import { Context } from "./context"
+import { Controller } from "./controller"
+import { TokenListObserver, TokenListObserverDelegate, Token, parseTokenString } from "@stimulus/mutation-observers"
+
+export class TargetObserver implements TokenListObserverDelegate {
+  readonly context: Context
+  readonly controller: Controller
+  readonly attributeName: string
+  private tokenListObserver: TokenListObserver
+
+  constructor(context: Context, controller: Controller) {
+    this.context = context
+    this.controller = controller
+    this.attributeName = `data-${context.identifier}-target`
+    this.tokenListObserver = new TokenListObserver(this.context.element, this.attributeName, this)
+  }
+
+  start() {
+    this.tokenListObserver.start()
+  }
+
+  stop() {
+    this.tokenListObserver.stop()
+  }
+
+  tokenMatched(token: Token): void {
+    if (this.controller.isConnected && this.containsDescendantWithToken(token.element, token.content)) {
```

Comment:
@sstephenson this PR adds the `isConnected` property to a controller to publish whether or not its `connect()` or `disconnect()` hooks have been invoked.

In this case, it's important to know so that callbacks are not fired during a controller's initial connection via the underlying `TokenListObserver.refresh()` call.